app/repositories/ScratchRepository.scala

```diff
@@ -45,17 +44,30 @@ class ScratchRepositoryImpl @Inject() (mongoComponent: ReactiveMongoComponent)
     )
     with ScratchRepository {
 
-  def save(process: JsObject): Future[UUID] = {
-    implicit val writer: OFormat[ScratchProcess] = ScratchProcessFormatter.mongoFormat
+  def save(process: JsObject): Future[RequestOutcome[UUID]] = {
     val document = ScratchProcess(UUID.randomUUID(), process)
-    collection.insert(ordered = false).one(document).map(_ => document.id)
+    insert(document)
+      .map { _ =>
+        Right(document.id)
+      }
+      .recover {
+        case e =>
+          logger.warn(e.getMessage)
+          Left(Errors(DatabaseError))
```

Comment:
Is it worthwhile using the DatabaseError here as it is replaced by internal server error in the scratch service.